### ISSUE TITLE
DataList examples and a11y documentation

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-next#1114]
Currently, some screen readers like JAWS and NVDA don't provide any indication when navigating between list items in the DataList component. A simple way to provide better separation of list items is to define the primary content section as a heading. 

As part of this issue, the DataList component should include additional examples and documentation so that consumers understand how to improve screen reader accessibility:

1. New example using `<h2>` element
    - Make a duplicate of Data List Simple, and use `<h2>` as the element for either the `div.pf-c-data-list__cell` element or the child `<span>` of that `<div>`
    - Include an Accessibility documentation section for that example that explains the importance of including headers in the DataList component when more than a primary content section is included. For example:
      >  When a list item includes more than one block of content, it can be difficult for some screen reader users to discern where one list item ends and the next one begins. A simple way to provide  better separation of list items is to define the primary content section as a heading. Headings are useful for indicating a new section of contents, but also provide an easy way for screen reader users to navigate to specific sections on the page. The heading level should be based on the context in which the DataList component is being used. For example, if the heading for the section that contains the DataList is a level 3, then `<h4>` elements should be used in the DataList list items. 
2. New example using just `<li>` elements
    - It's possible to create a datalist that doesn't include the wrapping elements inside the `<li>`: `<div class="pf-c-data-list__cell">`. Create a new example that illustrates this.

**Comment by jgiardino** [comment]
@mattnolting @mcoker I think the second item in the description was based on a recommendation from one of you? But I think the css was probably updated since then. I'm not seeing the data list display as expected if I just wrap contents with `<li class="pf-c-data-list__item">` without the additional classes inside that element. Do you remember this case coming up? And is there any expectation that this use case should be possible, or should I just remove it?

**Comment by mcoker** [comment]
I don't recall recommending a use case of the data list using `li.pf-c-data-list__item` elements on their own, but currently the wrapping elements inside are required to achieve proper spacing/layout, so I think you could remove that use case.

**Comment by mattnolting** [comment]
@jgiardino Agree w/@mcoker, although it does seem reasonable to include a fallback. I think remove the case for now..

**Comment by rachael-phillips** [comment]
We'll need to create a follow-up issue in React for this one
